src-self-hosted/cgen.zig

```diff
@@ -116,6 +124,12 @@ pub fn generate(file: *C, decl: *Decl) !void {
                             if (call.func.cast(ir.Inst.Constant)) |func_inst| {
                                 if (func_inst.val.cast(Value.Payload.Function)) |func_val| {
                                     const target = func_val.func.owner_decl;
+                                    const target_ty = target.typed_value.most_recent.typed_value.ty;
+                                    const ret_ty = target_ty.fnReturnType().tag();
+                                    if (ret_ty != .void and ret_ty != .noreturn) {
```

Comment:
I believe this should be 
```suggestion
                                    if (ret_ty.hasCodeGenBits()) {
```

Reply:
I'll double check; you're probably right.